From b12d5d818672bcd4e4dfacccb80b23555af4d873 Mon Sep 17 00:00:00 2001
From: Xisco Fauli <xiscofauli@libreoffice.org>
Date: Thu, 25 Jun 2020 21:18:48 +0200
Subject: [PATCH] UITest_writer_tests7: wait until the child is available
 (blind fix)

it fails spuriously with

Traceback (most recent call last):
File "sw/qa/uitest/writer_tests7/tdf99711.py", line 43, in test_tdf99711
self.assertEqual(get_state_as_dict(xWriterEdit.getChild('selectwidth'))['Text'], '10.00 mm')
uno.com.sun.star.uno.RuntimeException: Could not find child with id: selectwidth

Change-Id: If80dac67880a1f4200331a7894ba2ecae4504bb2
---
 sw/qa/uitest/writer_tests7/tdf99711.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sw/qa/uitest/writer_tests7/tdf99711.py b/sw/qa/uitest/writer_tests7/tdf99711.py
index 8540454cbb7d..1a6b7f069392 100644
--- a/sw/qa/uitest/writer_tests7/tdf99711.py
+++ b/sw/qa/uitest/writer_tests7/tdf99711.py
@@ -40,7 +40,11 @@ class tdf99711(UITestCase):
         self.xUITest.executeCommand(".uno:Sidebar")
         xWriterEdit.executeAction("SIDEBAR", mkPropertyValues({"PANEL": "TextPropertyPanel"}))
 
+        #wait until the sidebar is available
+        self.ui_test.wait_until_child_is_available(xWriterEdit, 'selectwidth')
         self.assertEqual(get_state_as_dict(xWriterEdit.getChild('selectwidth'))['Text'], '10.00 mm')
+
+        self.ui_test.wait_until_child_is_available(xWriterEdit, 'selectheight')
         self.assertEqual(get_state_as_dict(xWriterEdit.getChild('selectheight'))['Text'], '10.00 mm')
 
         self.xUITest.executeCommand(".uno:Sidebar")
-- 
2.20.1

